File: sdk/python/hitechcloud/resources/collocation.py
"""Collocation Services resource (3 endpoints)"""

from .base import BaseResource

class CollocationResource(BaseResource):
    """Collocation Services: PDU port management"""

    def list_pdu_ports(self, service_id: int) -> dict:
        """GET /api/service/{id}/pdu - List PDU ports assigned to service"""
        return self._http.get(f"/api/service/{service_id}/pdu")

    def get_pdu_port_state(self, service_id: int, port: int) -> dict:
        """GET /api/service/{id}/pdu/{port} - Get PDU port/outlet state"""
        return self._http.get(f"/api/service/{service_id}/pdu/{port}")

    def set_pdu_port_status(self, service_id: int, port: int, **kwargs) -> dict:
        """POST /api/service/{id}/pdu/{port} - Set PDU port/outlet state"""
        return self._http.post(f"/api/service/{service_id}/pdu/{port}", kwargs)

File: sdk/ruby/lib/hitechcloud/resources/collocation.rb
module HiTechCloud
  module Resources
    class Collocation < Base

        # Get PDU info
        def pdu(id)
          @http.get("/api/service/%{id}/pdu" % { id: id })
        end

        # Get PDU port
        def pdu_port(id, port)
          @http.get("/api/service/%{id}/pdu/%{port}" % { id: id, port: port })
        end

        # Toggle PDU port
        def toggle_pdu_port(id, port, data)
          @http.post("/api/service/%{id}/pdu/%{port}" % { id: id, port: port }, data)
      end
    end
  end
end
